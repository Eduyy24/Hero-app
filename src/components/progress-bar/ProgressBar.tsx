import { FunctionComponent } from "react";
import styles from './ProgressBar.module.css'
type Props = {
  max:number;
  value: number;
}
export const ProgressBar: FunctionComponent<Props> = ({max, value}) => {
  return (
    <div>
      <progress data-testid='progress' className={styles.progressBar} value={value} max={max} />
      <p className={styles.count}>{value} / {max}</p>
    </div>
  )

}