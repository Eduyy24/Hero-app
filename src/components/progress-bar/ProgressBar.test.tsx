import {render, screen} from '@testing-library/react';
import {ProgressBar} from './ProgressBar'

describe('<GeneralButton />', () => {
  test('Show ProgressBar correctly', () => {
    const max = 10
    const value = 9;

    render(<ProgressBar value={value} max={max} />)
    const progress = screen.getByTestId('progress');

    expect(progress).toMatchSnapshot()
    expect(progress).toHaveAttribute('max', max.toString())
    expect(progress).toHaveAttribute('value', value.toString())
    expect(progress).toBeInTheDocument()
  })
})