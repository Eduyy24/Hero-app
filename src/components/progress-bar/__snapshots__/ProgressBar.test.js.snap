// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneralButton /> Show ProgressBar correctly 1`] = `
<progress
  class="progressBar"
  data-testid="progress"
  max="10"
  value="9"
/>
`;
